students/mattmaeda/session11/test_generator.py

```diff
@@ -0,0 +1,78 @@
+"""
+test_generator.py
+
+tests the solution to the generator lab
+
+can be run with py.test or nosetests
+"""
+
+import generator_solution as gen
+
+
+def test_intsum():
+
+    g = gen.intsum()
+
+    assert next(g) == 0
+    assert next(g) == 1
+    assert next(g) == 3
+    assert next(g) == 6
+    assert next(g) == 10
+    assert next(g) == 15
+
+
+def test_intsum2():
+
+    g = gen.intsum2()
+
+    assert next(g) == 0
+    assert next(g) == 1
+    assert next(g) == 3
+    assert next(g) == 6
+    assert next(g) == 10
+    assert next(g) == 15
+
+
+def test_doubler():
+
+    g = gen.doubler()
+
+    assert next(g) == 1
+    assert next(g) == 2
+    assert next(g) == 4
+    assert next(g) == 8
+    assert next(g) == 16
+    assert next(g) == 32
+
+    for i in range(10):
+        j = next(g)
+
+    assert j == 2**15
+
+
+def test_fib():
+    g = gen.fib()
+
+    assert next(g) == 0
+    assert next(g) == 1
+    assert next(g) == 1
+    assert next(g) == 2
+    assert next(g) == 3
+    assert next(g) == 5
+    assert next(g) == 8
+    assert next(g) == 13
+    assert next(g) == 21
+
+
+def test_prime():
```

Comment:
did this test pass? it looked like your prime check only went to 7 -- so would not have returned the ones after 11 correctly -- though maybe you get to 51??

Reply:
Try all of them up to 100:

2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47, 53, 59, 61, 67, 71, 73, 79, 83, 89, 97